target_game_3.0.py

```diff
@@ -0,0 +1,379 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed May 27 11:17:33 2020
+
+@author: Aly
+"""
+
+from random import randint
+from math import sqrt
+
+from psychopy.visual import Window, ImageStim, TextStim, Circle
+from psychopy.core import wait
+from psychopy.event import getKeys, Mouse
+from psychopy.clock import CountdownTimer, Clock
+from psychopy import gui
+
+
+# I create a function which randomly changes the position of my target
+# I chose numbers depending on my DISP size (I want the target to be always
+# presented inside the screen), then the new position is set to the target, 
+# so the target can be moved
+def move_target_at_random_pos(target):
+    pos_x = randint(-700, 700)
+    pos_y = randint(-400, 400)
+    
+    new_pos = (pos_x, pos_y)
+    target.setPos(new_pos)
+    
+
+# Now I create a function that will draw 3 targets - one for each level of the game
+# On these targets I want to show where the player clicked (scored) on each level
+def draw_overview_target(target_x, target_y, mouse_clicks):
+    target = ImageStim(win, 'target.png', size = 420, pos=(target_x, target_y))
+    target.draw()
+    
+    for target_hit_pos in mouse_clicks:
+        if target_hit_pos['mouse_in_target']:
+            circle = Circle(win, radius=5, fillColor='yellow', lineColor='yellow')
+            circle.setPos((target_hit_pos['mouse_x']+target_x, target_hit_pos['mouse_y']+target_y))
+            circle.draw()
+
+# Since psychopy reads my target image as a square and not a circle, sometimes
+# a hit outside of the target is recorded as inside the target (as it is
+# inside the image). Therefore, we want to ensure that a point is given 
+# only when the player succesfully hits the target. We use the radius of the 
+# circle to find the distance between the centre of the target and the hit
+def distance_between_two_points(ax, ay, bx, by):
+    return sqrt((ay - by)**2 + (ax - bx)**2)
+
+
+def mouse_clicked_in_target(mouse, target):
+    mouse_x, mouse_y = mouse.getPos()
+    target_x, target_y = target.pos
+    # size returns width and height, since the target image is
+    # a square, it doesn't matter which one we pick
+    radius = target.size[0] / 2
+    distance = distance_between_two_points(mouse_x, mouse_y, target_x, target_y)
+    
+    return distance <= radius # we make sure the distance is not longer than
+                              # the radius
+
+# Now I create a function that will help us save these results in a csv file
+def write_results(target_hits_per_level, filename):
+    with open(filename, 'w') as file: # when with open --> file is closed automatically at the end

```

Comment:
Python style "Inline comments should be separated by at least two spaces from the statement."
```suggestion
    with open(filename, 'w') as file:  # when with open --> file is closed automatically at the end

```